File: ikaros/0151-reverse-words-in-a-string/reverse_words.py
class Solution:
    def reverseWords(self, s: str) -> str:
        arr = s.strip().split()
        return " ".join([word for word in arr[::-1]])
    

    


if __name__ == "__main__":
    ms = "a good   example"
    print(Solution().reverseWords(ms))

File: ikaros/0151-reverse-words-in-a-string/reverse_words.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(reverseWords("  hello world!  "))
}

func reverseWords(s string) string {
	array := strings.Split(strings.TrimSpace(s), " ")

	length := len(array)
	words := make([]string, 0, length)

	for i := length - 1; i >= 0; i-- {
		if len(array[i]) != 0 {
			words = append(words, array[i])
		}
	}

	return strings.Join(words, " ")
}
